splink/database_api.py

```diff
@@ -59,6 +67,8 @@ class DatabaseAPI(ABC, Generic[TablishType]):
 
     def __init__(self) -> None:
         self._intermediate_table_cache: CacheDictWithLogging = CacheDictWithLogging()
+        # TODO: replace this:
+        self._cache_uid: str = str(random.choice(range(10000)))
```

Comment:
This is interesting - e.g. the user could:
1. pass the same `db_api` to profile column as they do to the linker
2. pass different `db_api`s to the function

Given the possibiilty of (1), a new uuid would probably be needed when it's passed into the linker to protect against the cache being accidentally hit.  (would be very unlikely, exactly the same sql query would have to be run against different data, because the caching mechansim only checks the sql string, not whether the data has changed).

Doesnt seem too bad though - should be straightforward to re-assign the UID when the db_api is passed into linker